### PR TITLE
Auto/fix obstacle stop planner

### PR DESCRIPTION
**Pull request by Sugatyon**
## Description

change trajectory to trajectory points(vector of trajectory point) in function

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by Sugatyon** [comment]
@tkimura4 
cppcheckでfailしてしまうのですが、原因わかります？？
↓らへんが良くないと思うのですが、原因わからず、、、
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/678/files#diff-9493ac6ff50c82cecb25fb47d2f6fb1a6a512d07e8aad7d2693d94cc419a1107L1014-L1046

**Comment by tkimura4** [comment]
```
Starting >>> obstacle_stop_planner
--- stderr: obstacle_stop_planner
--- output: obstacle_stop_planner
Errors while running CTest
```

```
2: -- run_test.py: invoking following command in '/__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner':
2:  - /opt/ros/galactic/bin/ament_cppcheck --xunit-file /__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/build/obstacle_stop_planner/test_results/obstacle_stop_planner/cppcheck.xunit.xml --include_dirs /__w/AutowareArchitectureProposal.iv/AutowareArchitectureProposal.iv/planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/include
2: [src/node.cpp:1033]: (error: invalidContainerReference) Reference to output that may be invalid.
2: [src/node.cpp:1055]: (error: invalidContainerReference) Reference to output that may be invalid.
2: 2 errors
```

**Comment by tkimura4** [comment]
ちょっと調べてみます

File: planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp [hunk 1]
```diff
@@ -1098,105 +1107,105 @@ TrajectoryPoint ObstacleStopPlannerNode::getExtendTrajectoryPoint(
...
-  pcl::PointCloud<pcl::PointXYZ>::Ptr output_points_ptr)
+  pcl::PointCloud<pcl::PointXYZ>::Ptr output_points_ptr,
+  const std_msgs::msg::Header & trajectory_header)
 {
   // transform pointcloud
   geometry_msgs::msg::TransformStamped transform_stamped{};
   try {
     transform_stamped = tf_buffer_.lookupTransform(
-      trajectory.header.frame_id, input_points_ptr->header.frame_id, input_points_ptr->header.stamp,
+      trajectory_header.frame_id, input_points_ptr->header.frame_id, input_points_ptr->header.stamp,
```
Comment (Sugatyon) [inline planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp:1204]: ここtrajectoryのinputにheader含まれないので、"base_link"に変更しないとですね。

**Comment by tkimura4** [comment]
> cppcheckでfailしてしまうのですが、原因わかります？？
↓らへんが良くないと思うのですが、原因わからず、、、

reference (&)を使ってはいけないところで使っていたのが原因っぽかったです。
const auto & x = ...が変数のrename目的なら良いですが, 計算結果を格納してしまうとinvalid Refernceとなってしまいます

File: planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp (hunk 1 above)
Comment (Sugatyon) [reply planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp:1204]: pathCallbackのはじめでoutput_trajectoryにinput_msgを受け取って
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/678/files#diff-9493ac6ff50c82cecb25fb47d2f6fb1a6a512d07e8aad7d2693d94cc419a1107L566
関数searchPointcloudNearTrajectoryにoutput_trajectoryのheaderを引数として渡しているので現状で問題ないかと！
https://github.com/tier4/AutowareArchitectureProposal.iv/pull/678/files#diff-9493ac6ff50c82cecb25fb47d2f6fb1a6a512d07e8aad7d2693d94cc419a1107L1186-R1204

File: planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp [hunk 2]
```diff
@@ -586,26 +591,28 @@ void ObstacleStopPlannerNode::pathCallback(const Trajectory::ConstSharedPtr inpu
     resetExternalVelocityLimit();
   }
 
-  path_pub_->publish(output_trajectory);
+  path_pub_->publish(autoware_utils::convertToTrajectory(output_trajectory_points));
```
Comment (tkimura4) [inline planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp:594]: pubする前に
output.header = path.header;
みたいなheaderのコピーが必要ですね

Comment (Sugatyon) [reply planning/scenario_planning/lane_driving/motion_planning/obstacle_stop_planner/src/node.cpp]: inputのheaderをoutputのheaderにコピーしました！https://github.com/tier4/AutowareArchitectureProposal.iv/pull/678/commits/353da65f16d6a457d90b39eda4fe2fb29235a55d

**Review by tkimura4 — APPROVED** [review]
LGTM

**Pull request merged**